### ISSUE TITLE
Ability to use namespace name in objects created by NamespaceConfig CR 

### ISSUE THREAD

**Issue by redmikhail** [redhat-cop/namespace-configuration-operator#28]
Similar to issue #20 but for  OperatorGroup objects for OLM. OperatorGroup object is required for operator deployment using OLM .In cases when operator is intended to be deployed to single namespace one of the parameters that needs to be specified is ` targetNamespaces` which would be unknown until project(namespace) has been created and labeled , eg. 
``` apiVersion: operators.coreos.com/v1
kind: OperatorGroup
metadata:
  name: test-op-group
spec:
  serviceAccountName: test-sa-scoped
  targetNamespaces:
  - ${TARGET_NAMESPACE} 
```
While OperatorGroup can utilize selectors for target namespaces, it only works for one namespace. If more than one project is labeled OLM assumes that OperatorGroup is for multi-namespace operator deployment instead of creating multiple single-namespace OperatorGroup objects. In this case operators that are specifically designed to work in one namespace would fail to deploy with error similar to  `MultiNamespace InstallModeType not supported, cannot configure to watch 2 namespaces`  

It would be useful if we can have predefined variable for namespace that would be dynamically populated before creating the object. 

The use case for this functionality is multi-tenant OCP clusters where user potentially doesn't have cluster-admin privileges required to create OperatoGroup object to deploy operator

**Comment by raffaelespazzoli** [comment]
Duplicate of #20 
Duplicate of #29

**Comment by raffaelespazzoli** [comment]
this has been closed in #37
